src/skeleplex/graph/skeleton_graph.py

```diff
@@ -509,10 +605,122 @@ def sample_volume_slices_from_spline(
                     grid_spacing=(pixel_size, pixel_size),
                     sample_interpolation_order=0,
                     image_voxel_size_um=image_voxel_size_um,
+                    approx=approx,
                 )
                 segmentation_slice_dict[(u, v)] = segmentation_slice
 
         if segmentation is not None:
             return image_slice_dict, segmentation_slice_dict
         else:
             return image_slice_dict
+
+    def sample_volume_slices_from_spline_parallel(
```

Comment:
Please add a test. It can be simple (e.g., a line along the z axis with z slices of different constant values). I would be sure to use a pixel size that isn't 1 to make sure the conversions are being done correctly. You may need to set workers to 1 to make it work on CI.

Reply:
added a test for the normal sampling and the spline sampling